cypress/e2e/signIn.cy.js

```diff
@@ -3,32 +3,63 @@
 
 import SignInPageObject from '../support/pages/signIn.pageObject';
 import HomePageObject from '../support/pages/home.pageObject';
+import faker from 'faker';
 
 const signInPage = new SignInPageObject();
 const homePage = new HomePageObject();
 
+const wrongData = {
+  email: faker.internet.email().toLowerCase(),
+  password: faker.random.alphaNumeric(8)
+};
+
 describe('Sign In page', () => {
   let user;
 
-  before(() => {
+  beforeEach(() => {
     cy.task('db:clear');
     cy.task('generateUser').then((generateUser) => {
       user = generateUser;
+    cy.register(user.email, user.username, user.password);
+    signInPage.visit();
     });
   });
 
   it('should provide an ability to log in with existing credentials', () => {
-    signInPage.visit();
-    cy.register(user.email, user.username, user.password);
-
     signInPage.typeEmail(user.email);
     signInPage.typePassword(user.password);
     signInPage.clickSignInBtn();
 
     homePage.assertHeaderContainUsername(user.username);
   });
 
-  it('should not provide an ability to log in with wrong credentials', () => {
+  it('should not provide an ability to log in with wrong email', () => {
+    signInPage.typeEmail(wrongData.email);
+    signInPage.typePassword(user.password);
+    signInPage.clickSignInBtn();
+
+    signInPage.assertModalContainText();
+  });
+
+  it('should not provide an ability to log in with wrong password', () => {
+    signInPage.typeEmail(user.email);
+    signInPage.typePassword(wrongData.password);
+    signInPage.clickSignInBtn();
+
+    signInPage.assertModalContainText();
+  });
+
+  it('should not provide an ability to log in with an empty email field', () => {
+    signInPage.typePassword(user.password);
+    signInPage.clickSignInBtn();
+
+    signInPage.assertModalContainText();
```

Comment:
hard to understand what text should modal contain, need to add a string argument to this function like assertModalContainText('Email field is empty')